test/socket.js

```diff
@@ -41,6 +41,13 @@ describe('socket', function(){
   it('should support options', function(){
     sock = zmq.socket('req', { backlog: 30 });
     sock.backlog.should.equal(30);
+	// ZMTP 3.1 heartbeat
+	sock.heartbeat_ivl = 42;
+    sock.heartbeat_ivl.should.equal(42);
+	sock.heartbeat_ttl = 43;
+    sock.heartbeat_ttl.should.equal(43);
+	sock.heartbeat_timeout = 44;
+    sock.heartbeat_timeout.should.equal(44);
```

Comment:
Same wrong indentation here.
Also the wrong test block, as there should be given options in an object:
```js
sock = zmq.socket('req', {
  backlog: 30,
  heartbeat_ivl: 42,
});
```
You could try the `setsockopt()` function and see if the build error still persists. I cannot say atm why the second assignment fails here.
```js
sock.setsockopt(zmq.ZMQ_HEARTBEAT_IVL, 42);
sock.setsockopt(zmq.ZMQ_HEARTBEAT_TTL, 43);
sock.setsockopt(zmq.ZMQ_HEARTBEAT_TIMEOUT, 44);

// Test the options with the getsockopt() function
sock.getsockopt(zmq.ZMQ_HEARTBEAT_TTL).should.equal(43);
// ...

sock.setsockopt('heartbeat_ivl', 42);
sock.setsockopt('heartbeat_ttl', 43);
sock.setsockopt('heartbeat_timeout', 44);

// Same tests here
```